partner_communication/wizards/generate_communication_wizard.py

```diff
@@ -129,58 +116,48 @@ def _compute_progress(self):
     ##########################################################################
     #                             VIEW CALLBACKS                             #
     ##########################################################################
-    @api.onchange("selection_domain", "force_language")
+    @api.onchange("selection_domain")
     def onchange_domain(self):
-        if self.force_language and not self.language_added_in_domain:
-            domain = self.selection_domain or "[]"
-            domain = domain[:-1] + f", ('lang', '=', '{self.force_language}')]"
-            self.selection_domain = domain.replace("[, ", "[")
-            self.language_added_in_domain = True
         if self.selection_domain:
             self.partner_ids = self.env["res.partner"].search(
                 safe_eval(self.selection_domain)
             )
-        if not self.force_language:
-            self.language_added_in_domain = False
 
-    @api.onchange("model_id")
-    def onchange_model_id(self):
+    @api.onchange("model_id", "force_language")
+    def onchange_model_or_language(self):
         if self.model_id:
+            config = self.model_id.email_template_id
+            if self.force_language:
+                config.with_context(lang=self.force_language)
```

Comment:
in order to work, you need to reload the variable : `config = config.with_context()`